app/services/record_participant_declaration.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+require "json_schema/validate_body_against_schema"
+
+class RecordParticipantDeclaration
+  attr_accessor :params
+
+  class << self
+    def call(params)
+      new(params).call
+    end
+
+    def required_params
+      %i[participant_id lead_provider declaration_type declaration_date raw_event]
+    end
```

Comment:
Do we ever use this, other than in the instance method?

Reply:
It can be called externally too, which is why it's a class level public. It can be a useful pattern for an intermediate service to follow when it is called from a controller and wants to persist the passed in data.